api/src/test/java/org/openmrs/module/appointments/dao/impl/AppointmentDaoImplIT.java

```diff
@@ -306,4 +316,45 @@ public void shouldReturnAppointmentsWithoutDates() {
         assertNotNull(appointments);
         assertEquals(3, appointments.size());
     }
+
+    @Test
+    public void shouldReturnAppointmentWithEncounters() {
+        String appointmentUuid="75504r42-3ca8-11e3-bf2b-0800271c13349";
+        Appointment appointment = appointmentDao.getAppointmentByUuid(appointmentUuid);
+        assertNotNull(appointment);
+        assertEquals(2, appointment.getFulfillingEncounters().size());
+    }
+
+    @Test
+    public void voidingAppointmentShouldNotVoidFulfillingEncounters() {
+        String appointmentUuid="75504r42-3ca8-11e3-bf2b-0800271c13349";
+        Appointment appointment = appointmentDao.getAppointmentByUuid(appointmentUuid);
+        appointment.setVoided(true);
+        appointmentDao.save(appointment);
+        Context.flushSession();
+        Context.clearSession();
+
+        appointment = appointmentDao.getAppointmentByUuid(appointmentUuid);
+        assertTrue(appointment.getVoided());
+        Encounter enc1 = encounterService.getEncounterByUuid("f303e49f-24a9-41bb-810b-5d07b881c0e0");
+        assertFalse(enc1.getVoided());
+        Encounter enc2 = encounterService.getEncounterByUuid("785826c1-f137-4b4f-bb2f-07b5c546d139");
+        assertFalse(enc2.getVoided());
+    }
```

Comment:
So, after writing this, I don't know if this (or the new service test class and method I added) really test much, as I think the save handlers and void handlers depend on the Appointments Service being an "OpenMRS Service", which the Appointments Service is not. Also, there is no "voidAppointment" method at the API or DB level.  So I don't think there's much risk of voided getting cascaded down to the encounter, but hopefully (though probably doubtful) one of these new voiding tests will catch a problem if it gets introduced in the future.

Reply:
do we see a case, when an appointment (post fulfilment through an encounter gets) gets voided? Should this be allowed? If not, should we put check/validate and restrict voiding of appointment in such case?  if we do allow that - then should we remove the association with the encounter?